src/avram/upsert.cr

```diff
@@ -90,6 +90,24 @@ module Avram::Upsert
     end
   end
 
+  macro upsert_unique_on(*attribute_names)
+    def self.bulk_upsert(upserts)
```

Comment:
if we rename (and add types) to this and the below bulk_upsert, overloading will work exactly as expected, even with upsert_unique_on and the other upsert_lookup macro being split for now!